sigstore/_verify.py

```diff
@@ -148,6 +151,18 @@ def verify(
             output(f"Failed to validate Rekor entry's SET: {inval_set}")
             continue
 
+        # 6) Verify that the signing certificate was valid at the time of signing
+        integrated_time = datetime.datetime.utcfromtimestamp(entry.integrated_time)
+        if (
+            integrated_time < cert.not_valid_before
+            or integrated_time > cert.not_valid_after
```

Comment:
Just want to confirm that neither of these should be inclusive? Specifically what should happen when `integrated_time == cert.not_valid_before`?

Reply:
RFC 5280 isn't completely explicit about this, but I believe `notBefore` is inclusive and `notAfter` is exclusive. 

Quote https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.5:

> The field is represented as a SEQUENCE of two dates:
   the date on which the certificate validity period begins (notBefore)
   and the date on which the certificate validity period ends
   (notAfter).  Both notBefore and notAfter may be encoded as UTCTime or
   GeneralizedTime.

My reading of that is: "the certificate validity begins with `notBefore` (meaning inclusive) and ends with `notAfter` (meaning exclusive)".

Reply:
Whoops, my language there is confusing: `time == notBefore` is valid and `time == notAfter` is invalid, based on my interpretation.

Reply:
[d574759](https://github.com/trailofbits/sigstore-python/pull/42/commits/d574759123981512c7ab2f72172943b74c09931e)

Reply:
Good eye. Thanks.

Reply:
Looks like d574759 does the opposite of what I'd expect?

Reply:
Oh, this is the failure case. 👍